### PR TITLE
Fix compute resource selector bug

### LINKED ISSUE #148: On my system, the compute resource selection widget does not include hosts
On the second page of the VCH Creation Wizard, I would expect hosts to be displayed as children of the cluster, but they are not:
image

These hosts appear, as expected, in the Hosts & Clusters tree:
image

Let me know if there's any diagnostic or log information I could collect to aid in debugging.

### PR DESCRIPTION
**Pull request by jooskim**
Fixes #148

### CODE REVIEW

**Comment by jooskim** [comment]
After fix:
image

File: h5c/vic/src/vic-webapp/e2e/app.po.ts [hunk 1]
```diff
@@ -98,7 +98,8 @@ export class VicWebappPage {
   }
 
   selectComputeResource() {
-    this.waitForElementToBePresent(this.buttonComputeResource);
+    this.waitForElementToBePresent('button.clr-treenode-caret');
+    this.clickByCSS('button.clr-treenode-caret');
```
Comment (jak-atx) [inline h5c/vic/src/vic-webapp/e2e/app.po.ts]: can you make these selectors some variable, something semantic?

File: h5c/vic/src/vic-webapp/src/app/create-vch-wizard/compute-capacity/compute-resource-treenode.component.ts [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+/**
+ * Component that renders a tree view of the inventory items on the selected Datacenter
+ * where Clusters, ClusterHostSystems and sstandalone hosts are displayed and selectable
+ */
+
+@Component({
+  selector: 'vic-compute-resource-treenode',
+  styleUrls: ['./compute-capacity.scss'],
+  template: `
```
Comment (jak-atx) [inline h5c/vic/src/vic-webapp/src/app/create-vch-wizard/compute-capacity/compute-resource-treenode.component.ts]: save this into a template file please

**Comment by jooskim** [comment]
Regarding the e2e test broken, it seems that it hangs in the storage capacity section indefinitely and times out. Is it only happening in the protractor test?

**Comment by jooskim** [comment]
Also check if you have slow connection

**Comment by jak-atx** [comment]
I don't see it happen on my download cert branch and it's same connection. I'll try your branch again.

**Comment by jooskim** [comment]
Sure, it seems strange that the spinner stays there in the storage capacity section, as usually i don't see it at all. Something might've gone wrong.

File: h5c/vic/src/vic-webapp/e2e/app.po.ts [hunk 3]
```diff
@@ -141,13 +143,13 @@ export class VicWebappPage {
...
 
   deleteVch(vch) {
-    browser.ignoreSynchronization = true;
     this.waitForElementToBePresent(this.actionBar + vch);
     this.clickByCSS(this.actionBar + vch);
     this.clickByCSS('button.' + vch);
     browser.switchTo().defaultContent();
     this.waitForElementToBePresent(this.iframeModal);
     this.switchFrame(this.iframeModal);
+    this.waitForElementToBePresent(this.labelDeleteVolumes);
```
Comment (zjs) [inline h5c/vic/src/vic-webapp/e2e/app.po.ts]: In the future, it'd probably be better to put unrelated changes into their own PR.

File: h5c/vic/src/vic-webapp/src/app/create-vch-wizard/compute-capacity/compute-capacity.component.ts [hunk 4]
```diff
@@ -1,3 +1,5 @@
+import 'rxjs/add/observable/timer';
```
Comment (zjs) [inline h5c/vic/src/vic-webapp/src/app/create-vch-wizard/compute-capacity/compute-capacity.component.ts]: Is this actually intended to go about the copyright header?

File: h5c/vic/src/vic-webapp/src/app/create-vch-wizard/create-vch-wizard.component.html [hunk 5]
```diff
@@ -65,17 +65,6 @@
          target="_blank">
         <clr-icon shape="help" class="is-info"></clr-icon>
       </a>
-      <clr-alert *ngIf="errorFlag"
```
Comment (zjs) [inline h5c/vic/src/vic-webapp/src/app/create-vch-wizard/create-vch-wizard.component.html]: What's the purpose of the changes in this file?

Comment (jooskim) [reply h5c/vic/src/vic-webapp/src/app/create-vch-wizard/create-vch-wizard.component.html]: We should've filed a separate issue for this, but it's a cosmetic issue where the Clarity warning box, if any error is present, was appearing twice in the view. It's a bit of an excuse to say we wanted to sneak in some small stuff as we go, but will improve next time.

File: h5c/vic/src/vic-webapp/src/app/create-vch-wizard/compute-capacity/compute-capacity.component.ts (hunk 4 above)
Comment (jooskim) [reply h5c/vic/src/vic-webapp/src/app/create-vch-wizard/compute-capacity/compute-capacity.component.ts]: Ah, it's my VSCode extension that did it. I just turned it off completely, as it has been more of annoyance than help. Will address this too.

File: h5c/vic/src/vic-webapp/e2e/app.po.ts (hunk 3 above)
Comment (jooskim) [reply h5c/vic/src/vic-webapp/e2e/app.po.ts]: I think it makes sense to update E2E test case logics at the same time especially when you updated a component or service that might affect the way the UI is behaving. This PR is going to cause a spinner to spin until clusters and standalone hosts are loaded, which was not pretty much the case with the status quo, which was already in a broken state but somehow failed silently.

**Comment by jooskim** [comment]
@zjs For that I would definitely prefer to build a custom OVA and send it to @lgayatri. I think I did successfully built once before, but it consistently fails now.

**Comment by jooskim** [comment]
@zjs And you're right that I've removed the cache from the Angular service method that loads compute resources, as currently there's not much value in doing that. Once wizard pages are loaded the views and states are retained unless you close the wizard, which means I don't need to implement the cache. It may even cause unexpected states.

**Comment by jak-atx** [comment]
Tested with latest commit and e2e tests passing. lgtm.

**Comment by lgayatri** [comment]
@jooskim If this fix will not be available in tonight's RC3, then please ping me the custom OVA build.
